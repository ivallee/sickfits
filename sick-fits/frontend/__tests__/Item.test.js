import Item from '../components/Item';
import { shallow } from 'enzyme';

const fakeItem = {
  id: 'abc123',
  title: 'a dope item',
  price: 5400,
  description: 'this is an item that exists',
  image: 'item.jpg',
  largeImage: 'largeItem.jpg',
};

describe('<Item/>', () => {
  it('renders the image', () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    const img = wrapper.find('img');
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it('renders price, title and description', () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    const PriceTag = wrapper.find('PriceTag');
    expect(PriceTag.children().text()).toBe('$54');
    expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
    expect(wrapper.find('p').text()).toBe(fakeItem.description);
  });

  it('renders the buttons', () => {
    const wrapper = shallow(<Item item={fakeItem} />);
    const buttonList = wrapper.find('.buttonList');
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find('Link').exists()).toBe(true);
    expect(buttonList.find('DeleteItem').exists()).toBe(true);
  });

});