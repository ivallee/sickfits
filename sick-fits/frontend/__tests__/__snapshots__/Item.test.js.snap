// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="a dope item"
    src="item.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "abc123",
          },
        }
      }
    >
      <a>
        a dope item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $54
  </PriceTag>
  <p>
    this is an item that exists
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "abc123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="abc123"
    />
    <DeleteItem
      id="abc123"
    >
      Delete
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
